tests/integration/suite/components/versionv2.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	procdaprd "github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(versionv2))
+}
+
+// versionv2 tests dapr will load v1 and v2 components.
+type versionv2 struct {
+	proc *procdaprd.Daprd
+}
+
+func (v *versionv2) Setup(t *testing.T) []framework.Option {
+	v.proc = procdaprd.New(t, procdaprd.WithComponentFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: etcd-v1
+spec:
+  type: state.etcd
+  version: v1
```

Comment:
What if version is omitted? Can you add a scenario for that too?

Reply:
@artursouza It failed unless we register with no version (which I've added). I've made the default version `v1` for backwards compat. Let me know if you want me to change to `v2`/

Reply:
No, default should be `v1` for backwards-compat. Otherwise all users who did not specify a version (and there's probably a lot) would get upgraded.

Reply:
> No, default should be `v1` for backwards-compat. Otherwise all users who did not specify a version (and there's probably a lot) would get upgraded.

Agree.